crates/pubnub_transport_reqwest/src/reqwest.rs

```diff
@@ -13,9 +14,12 @@ struct TransportReqwest {
 #[async_trait::async_trait]
 impl Transport for TransportReqwest {
     async fn send(&self, request: TransportRequest) -> Result<TransportResponse, PubNubError> {
+        let path = prepare_path(&request.path, &request.query_parameters);
+        let request_url = format!("{}{}", &self.hostname, path);
```

Comment:
Maybe we can move making the request_url into the `prapare_path` function and rename it to `prepare_url`.